pkg/resource/snapshot/utility.go

```diff
@@ -30,6 +37,15 @@ func isDeleting(r *resource) bool {
 	return status == deleteStatus
 }
 
+// isNotReadyForDeleting returns true if supplied cluster resource state is 'copying', 'creating', or 'restoring'
+func isNotReadyForDeleting(r *resource) bool {
+	if r == nil || r.ko.Status.Status == nil {
+		return false
+	}
+	status := *r.ko.Status.Status
+	return status == copyingStatus || status == creatingStatus || status == restoringStatus
```

Comment:
I would do this the other way around. Check if `status != available && status != failed` - since that's what the service will check anyway

Reply:
There is an issue here. `InvalidSnapshotStateFault` shows `state` of `snapshot` has to be `available` or `failed`. However, when I check `describe-snapshot` API, `status` of `snapshot` doesn't have `failed` as an option. Hence, I'm not sure if there is `failed` `status` for `snapshot`. https://docs.aws.amazon.com/cli/latest/reference/memorydb/describe-snapshots.html
Also, if we do the another way, then it might should be `status != available && status != failed && status != deleting`

Reply:
```Status -> (string)

The status of the snapshot. Valid values: creating | available | restoring | copying | deleting.
```
`available` is there.

Reply:
I'm checking with my team for it now.

Reply:
It has "failed" as `status` for `snapshot`.